core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -243,7 +246,17 @@ public void configure() {
          if (rvrl != null) {
             rvrl.remoteValueNotFound(key);
          }
-         return null;
+         if (!hasSuccesfulResponse) {
+            RpcException rpcException = new RpcException("No successful response: " + responses);
+            for (Response r : responses.values()) {
+               if (r instanceof ExceptionResponse) {
+                  rpcException.addSuppressed(((ExceptionResponse) r).getException());
+               }
+            }
+            throw rpcException;
```

Comment:
This check should be done in `JGroupsTransport`, like we call `checkRsp` for the responses of synchronous RPCs.